torchtext/datasets/iwslt2016.py

```diff
@@ -261,12 +261,10 @@ def IWSLT2016(
     )
 
     cache_decompressed_dp = cache_compressed_dp.on_disk_cache(filepath_fn=lambda x: inner_iwslt_tar)
-    cache_decompressed_dp = (
-        FileOpener(cache_decompressed_dp, mode="b")
-        .load_from_tar()
-        .filter(lambda x: os.path.basename(inner_iwslt_tar) in x[0])
-    )
+    cache_decompressed_dp = cache_decompressed_dp.open_files(mode="b").load_from_tar()
+    cache_decompressed_dp = cache_decompressed_dp.filter(lambda x: os.path.basename(inner_iwslt_tar) in x[0])
     cache_decompressed_dp = cache_decompressed_dp.end_caching(mode="wb", same_filepath_fn=True)
+    cache_decompressed_dp_1, cache_decompressed_dp_2 = cache_decompressed_dp.fork(num_instances=2)
```

Comment:
The recommended pattern is to use `fork` when you need the same DataPipe for multiple usages later.